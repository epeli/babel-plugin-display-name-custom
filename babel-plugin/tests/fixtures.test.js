const babel = require("babel-core");
const fs = require("fs");

const fixtures = fs.readdirSync(__dirname + "/fixtures");


describe("compile fixtures to snapshots:", () => {

    fixtures.forEach(fixtureFile => {

        test(fixtureFile, () => {


            const lines = fs.readFileSync(__dirname + "/fixtures/" + fixtureFile).toString().split("\n");

            console.log(lines[0].slice(2).trim());
            const options = JSON.parse(lines[0].slice(2).trim());

            const output = babel.transform(lines.slice(1).join("\n"), {
                babelrc: false,
                plugins: [
                    [__dirname + "/../plugin.js", options],
                ],
            });

            expect(output.code).toMatchSnapshot();
        });

    });

});
