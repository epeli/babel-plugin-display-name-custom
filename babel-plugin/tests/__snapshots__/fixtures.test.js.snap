exports[`compile fixtures to snapshots: alternative-name.js 1`] = `
"import create from \"react-simple\";

const Foo = create(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`compile fixtures to snapshots: argument-shadow.js 1`] = `
"import simple from \"react-simple\";

function fn(simple) {
    var no = simple();
}

function outer(simple) {
    function inner() {
        var no = simple();
    }
}"
`;

exports[`compile fixtures to snapshots: assignment-only.js 1`] = `
"import simple from \"react-simple\";

var no = simple;"
`;

exports[`compile fixtures to snapshots: basic.js 1`] = `
"import simple from \"react-simple\";

const Foo = simple(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`compile fixtures to snapshots: custom.js 1`] = `
"import createComponent from \"other\";

const Foo = createComponent(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`compile fixtures to snapshots: declaration-shadow.js 1`] = `
"import simple from \"react-simple\";

function fn() {
    const simple = () => null;
    var no = simple();
}

function outer() {
    const simple = () => null;
    function inner() {
        var no = simple();
    }
}"
`;

exports[`compile fixtures to snapshots: export.js 1`] = `
"import simple from \"react-simple\";

export const Foo = simple(\"div\", {
    color: \"red\"
});
Foo.displayName = \"Foo\";"
`;

exports[`compile fixtures to snapshots: inside-function.js 1`] = `
"import simple from \"react-simple\";

function fn() {
    const Foo = simple(\"div\", {
        color: \"red\"
    });
    Foo.displayName = \"Foo\";
}

function outer() {
    function inner() {
        const Foo = simple(\"div\", {
            color: \"red\"
        });
        Foo.displayName = \"Foo\";
    }
}"
`;

exports[`compile fixtures to snapshots: named-export.js 1`] = `
"import { createComponent, no } from \"amodule\";

const Foo = createComponent(\"div\", {
    color: \"red\"
});

Foo.displayName = \"Foo\";
var nope = no();"
`;

exports[`compile fixtures to snapshots: named-export-as.js 1`] = `
"import { createComponent as other, no } from \"amodule\";

const Foo = other(\"div\", {
    color: \"red\"
});

Foo.displayName = \"Foo\";
var nope = no();"
`;
